src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java

```diff
@@ -97,6 +169,146 @@ private boolean checkIfExecute()
         return false;
     }
 
+    @Override
+    public void requestMaterialsIfRequired()
+    {
+        if (!Configurations.builderInfiniteResources)
+        {
+            requestMaterials();
+        }
+    }
+
+    /**
+     * Iterates through all the required resources and stores them in the building.
+     * Suppressing Sonar Rule Squid:S135
+     * The rule thinks we should have less continue and breaks.
+     * But in this case the rule does not apply because code would become unreadable and uneffective without.
+     */
+    @SuppressWarnings("squid:S135")
+    private void requestMaterials()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "requestMaterials" is 16 which is greater than 15 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)